### PR TITLE
No binary scala warning if 3 and 2.13

### LINKED ISSUE #6388: [1.5.0] Binary version warning between Scala 3.0.0 and 2.13
## steps

```scala
scalaVersion := "3.0.0-RC1"
libraryDependencies += "org.scala-lang" % "scala-reflect" % "2.13.4"
useCoursier := false
```

```shell
sbt:root> update
[warn] Binary version (2.13) for dependency org.scala-lang#scala-reflect;2.13.4
[warn]  in default#root_3.0.0-RC1;0.1.0-SNAPSHOT differs from Scala binary version in project (3.0.0-RC1).
[success] Total time: 0 s, completed Mar 12, 2021, 9:34:17 AM
```

## problem

sbt warns about the binary version of scala-reflect being 2.13.4

## expectation

no warning since 2.13.4 is binary compatible with 3.0.0-RC1

## notes

The log is coming from here:

https://github.com/sbt/librarymanagement/blob/b280c2b0b001f185b0cc59d395d8986838523495/ivy/src/main/scala/sbt/internal/librarymanagement/IvyScalaUtil.scala#L151-L155

### PR DESCRIPTION
**Pull request by adpi2**
Fix https://github.com/sbt/sbt/issues/6388

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=develop&io=develop&librarymanagement=pull/372/head&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

**Review by eed3si9n — APPROVED** [review]
Thanks!

File: ivy/src/main/scala/sbt/internal/librarymanagement/IvyScalaUtil.scala [hunk 1]
```diff
@@ -144,7 +144,13 @@ object IvyScalaUtil {
       val depBinaryVersion = CrossVersion.binaryScalaVersion(id.getRevision)
       def isScalaLangOrg = id.getOrganisation == scalaOrganization
       def isScalaArtifact = scalaArtifacts.contains[String](id.getName)
-      def hasBinVerMismatch = depBinaryVersion != scalaBinaryVersion
+
+      def hasBinVerMismatch =
+        depBinaryVersion != scalaBinaryVersion &&
+          // scala 2.13 is compatible with scala 3.x
+          Seq(depBinaryVersion, scalaBinaryVersion)
+            .forall(bv => bv.startsWith("3") || bv.startsWith("2.13"))
```
Comment (eed3si9n) [inline ivy/src/main/scala/sbt/internal/librarymanagement/IvyScalaUtil.scala:152]: Looks like dependency-management/scala-organization-version-check is failing. Maybe the logic is flipped here.

Comment (eed3si9n) [reply ivy/src/main/scala/sbt/internal/librarymanagement/IvyScalaUtil.scala:152]: https://github.com/sbt/librarymanagement/pull/373

**Pull request merged**